src/translator/service.h

```diff
@@ -110,18 +110,13 @@ class AsyncService {
     bool cacheEnabled{false};  ///< Whether to enable cache or not.
     size_t cacheSize{2000};    ///< Size in History items to be stored in the cache. Loosely corresponds to sentences to
```

Comment:
I doubt this comment. This is like theoretical max number of sentences. The chance 2000 sentences all have a different `hash % 2000` is pretty slim. In TranslateLocally I set it to 65536, and on Crime & Punishment it has about 1600 collisions on 15000 sentences. So occupancy is about 22%? Or see my earlier not very scientific exploration, https://github.com/XapaJIaMnu/translateLocally/pull/75#issuecomment-1004226518 

Edit: I have more of a problem with it in the description of the `--cache-size` option.

Reply:
We need a better hash function..... murmurhash perhpas

Reply:
That was my intuition as well. I tested it with wrapping the vocab offset size_t with `std::hash` to spread the bytes a bit and hopefully get fewer colissions, but it didn't have a real impact on the numbers I just mentioned. No idea whether that is because of the input I tried, or inherently the limit of how much of the cache can be reached by `N(0..vocab-size) % 65536` while being mixed up through a couple of hash combine functions, or something completely different. I think this (this seems to be my favourite closing line this evening) needs its own issue to explore it a bit further.

(I tested both std::hash<size_t> and murmurhash a couple of months ago when I was working on landing this in TranslateLocally. I didn't find any improvements back then, otherwise I would have opened an issue about it.)

Reply:
My problem here with this comment though is that 2000 seems to be some theoretical limit that is never reached. I'm afraid that downstream will read this and think "yeah no we're never going to have more than 2000 sentences that occur multiple times and are worth caching" and they end up with like 450 sentences in a cache and no guarantee that the more often occurring sentences are not constantly pushed out of the cache by collisions.

Reply:
We should definitely open an issue lest this goes forgotten.

Reply:
So there are two things here:

1. What does this parameter control: Strong correspondence is to History items. Weak correspondence is the number of sentences to cache. That's all that's meant by the comment. I see that random replacement will lead to troubling interpretations. We can expand the comment to be clearer about the nuances.
2. Reasonable default: 2000 vs 65536 - I haven't been able to experiment, I kind of wanted to expose cache hit/misses from Service and get this from somewhere. https://github.com/jerinphilip/tagtransfer/issues/3 is kept for the same.